pkg/network/netlink/bpf_sampler.go

```diff
@@ -0,0 +1,31 @@
+package netlink
+
+import (
+	"errors"
+	"math"
+
+	"golang.org/x/net/bpf"
+)
+
+var errInvalidSamplingRate = errors.New("sampling rate must be within (0, 1)")
+
+// GenerateBPFSampler returns BPF assembly for a traffic sampler
+func GenerateBPFSampler(samplingRate float64) ([]bpf.RawInstruction, error) {
```

Comment:
I think this would actually be a better fit for the `ebpf` package :)

Reply:
👍 I'll move it

Reply:
Unfortunately I can't do that. It would generate a cyclical dependency:
* `pkg/ebpf` -> `pkg/network/netlink`
* `pkg/network/netlink` -> `pkg/ebpf`